scripts/metadata_enrichment/tob_bioheart_sequencing_dates.py

```diff
@@ -0,0 +1,234 @@
+            for both bioheart and tob-wgs datasets
+    :rtype: defaultdict
+    """
+    # gives us json output from graphql query
+    # Results are seen in graphql interface as a dictionary
+    query_result_tob = query(QUERY_PROJECT_ASSAYS, variables={'datasetName': 'tob-wgs'})
+    query_result_bioheart = query(
+        QUERY_PROJECT_ASSAYS,
+        variables={'datasetName': 'bioheart'},
+    )
+
+    # list of dictionaries of samples per External ID
+    tob_samples = query_result_tob['project']['samples']
+    bioheart_samples = query_result_bioheart['project']['samples']
+
+    # Create default dicts for the two datasets: maps tube IDs to samples
+    tob_kccg_id = 'KCCG FluidX tube ID'
+    fluidx_to_assay_ids_tob = create_default_dict(tob_samples, tob_kccg_id)
+    bioheart_kccg_id = 'fluid_x_tube_id'
+    fluidx_to_assay_ids_bioheart = create_default_dict(
+        bioheart_samples,
+        bioheart_kccg_id,
+    )
+
+    return append_dictionaries(fluidx_to_assay_ids_tob, fluidx_to_assay_ids_bioheart)
+
+
+def create_default_dict(samples: list, kccg_id: str) -> defaultdict:
+    """
+    Creates defaultdict mapping kccg tube ids to all related samples.
+    Flexible and can be applied to tob-wgs and bioheart.
+
+    :return: defaultdict mapping kccg tubes to sample ids
+    :rtype: defaultdict(list)
+    """
+    assays = []
+    for sample in samples:
+        assays.extend(sample['assays'])
+
+    # Use default dict to group assays with fluid X tube metadata
+    tube_to_assays_defaultdict = defaultdict(list)
+    for assay in assays:
+        # per assay, extract assay ID and fluid X tubeID
+        # fluid tube id is the key; list contains assay IDs
+        fluidx_tube_id = assay['meta'].get(kccg_id)
+        if kccg_id.endswith('tube_id'):
+            tube_to_assays_defaultdict[fluidx_tube_id.split('_')[1]].append(
+                assay.get('id'),
+            )
+        else:
+            tube_to_assays_defaultdict[fluidx_tube_id].append(assay['id'])
+    return tube_to_assays_defaultdict
+
+
+def append_dictionaries(
+    fluidx_to_assay_ids_tob: defaultdict,
+    fluidx_to_assay_ids_bioheart: defaultdict,
+) -> defaultdict:
+    """
+    Append two defaultDict objects into a single dictionary.
+    Verifies that there are no intersecting tube ids
+
+    :return: When no intersection of keys
+    """
+    result = defaultdict(list)
+    dicts_to_merge = [fluidx_to_assay_ids_tob, fluidx_to_assay_ids_bioheart]
+    # First check to ensure that there are no overlapping keys
+    set_tob = set(fluidx_to_assay_ids_tob.keys())
+    set_bioheart = set(fluidx_to_assay_ids_bioheart.keys())
+
+    if len(set_tob.intersection(set_bioheart)) == 0:
+        # Append bioheart dict to tob dict
+        for d in dicts_to_merge:
+            for key, value in d.items():
+                result[key].append(value)
+        return result
+    else:  # noqa: RET505
```

Comment:
Since the `if` condition will always lead to a `return`, there is no need for an `else` here (hence the RET505 flake8 message).

Personally I think that it looks clearer with the `else` so feel free to leave it as is, but just in case it wasn't clear why the RET505 flag is appearing, it's because the else is considered "superfluous".

Reply:
I still want the else block. Suspect this has arisen from some lazy programming. Changed it to return at the end of the function instead.